StreamChatClient_v3/Controllers/ChannelListControllerPublishers.swift

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+@available(iOS 13, *)
+extension Client {
+    /// Creates a new `ChannelListController.Publishers` object with the provided channel query.
+    ///
+    /// - Parameter query: The query specify the filter and sorting of the channels the controller should fetch.
+    /// - Returns: A new instance of `ChannelListController.Publishers`.
+    ///
+    @available(iOS 13, *)
+    public func channelListControllerPublishers(query: ChannelListQuery) -> ChannelListControllerGeneric<ExtraData>.Publishers {
+        let controller = ChannelListControllerGeneric(query: query, client: self)
+        return .init(controller: controller)
+    }
+}
+
+/// Describes possible states of Controller's remote activity.
+public enum RemoteActivity {
+    /// The controller is not listening for updates. Call `startUpdating` to fetch the remote data and start listening for changes.
+    case none
+    
+    /// Controller actively tries to refresh the local data.
+    case fetchingRemoteData
+    
+    /// Controller is listening for incoming changes.
+    case listening
+    
+    /// Remote data fetch failed. Controller might still be able to show the recent incoming changes, but it's possible
+    /// the presented data are not complete.
+    case failed(error: Error?)
+}
+
+@available(iOS 13, *)
+extension ChannelListControllerGeneric {
+    /// An object which wraps `ChannelListController` and exposes its properties as Publishers.
+    public class Publishers {
+        /// Publishes changes to the remote activity state. You can use this publisher to show activity indicator to users
+        /// if the controller is fetching the remote data.
+        public var remoteActivityPublisher: AnyPublisher<RemoteActivity, Never> {
+            remoteActivity
+                // Is something like this the way to keep `Publishers` object alive if there is a subscriber?
+                .map { [self] in _ = self; return $0 }
```

Comment:
What's the issue this is resolving?

Reply:
If there is someone subscribed to the publisher, the `ChannelListControllerPublishers` object should be kept alive for the duration of the subscription. I wasn't sure about this part, but it seems to be working as expected 🤷‍♂️

Reply:
so... what happens if you remove this part? It's deallocated??

Reply:
Yes. I test for this explicitly in the tests where I "nil" the `publishers` object. If we don't find a better solution, we could create a custom operator with this functionality (like `own(...) or `keepAlive()`) that would do something like this.

Reply:
I see.
```swift
// Remove the strong reference to `Publishers` object. The chain should keep it alive.
publishers = nil
```
we want any created chain to keep a ref to the original publisher.
I think that's counter-intuitive and unexpected, we want to get rid of subscriptions when the publisher is deallocated., just like allowing delegates to be deallocated. Why did you choose that subscription chains should work even after publisher is deallocated?

Reply:
I think we don't understand each other. The behavior as implemented now is intuitive. For example, you want to be able to do this:
```
client
  .channelListControllerPublishers(id: xxx)
  .channels
  .sink { updateUI($0) }
  .store(in: &cancellables)
```
The `channelListControllerPublishers` objects should be alive and receiving updates until the chain is alive.

Reply:
I have the same question as @b-onc. I don't think you have to do that.